### PR TITLE
Fixed python3 bloom-release, json.loads encoding

### PR DESCRIPTION
**Pull request by gemarcano**
json.loads excepts a string, but resp.read() returns a bytes object in
Python3. Added a helper function to try to detect the encoding of the
response, and use that to decode the bytes to str.

Tested this briefly with Python2 to make sure it wasn't broken. Tried to make it compatible.

### CODE REVIEW

**Comment by nuclearsandwich** [comment]
Thanks for the contributed fix. Do you have an example of the error this addresses?

I don't believe there are any automated tests around this behavior. What was your test pattern while developing it?

**Comment by gemarcano** [comment]
This was the command I invoked:
```
python3 ~/.local/bin/bloom-release --rosdistro lunar --track lunar geographiclib-release --edit
```
Where that bloom-release file was installed from a built bloom package from this repository. geographiclib-release is a third party package I'm trying to integrate into the ROS environment. The error happened early enough in the process I don't think the package/repository matters.

The error I get is:
```
Traceback (most recent call last):
  File "/usr/lib/python-exec/python3.5/bloom-release", line 11, in <module>
    load_entry_point('bloom==0.6.1', 'console_scripts', 'bloom-release')()
  File "/usr/lib64/python3.5/site-packages/bloom/commands/release.py", line 1306, in main
    args.override_release_repository_push_url)
  File "/usr/lib64/python3.5/site-packages/bloom/commands/release.py", line 1152, in perform_release
    release_repo = get_release_repo(repository, distro, override_release_repository_url)
  File "/usr/lib64/python3.5/site-packages/bloom/commands/release.py", line 354, in get_release_repo
    url = get_repo_uri(repository, distro)
  File "/usr/lib64/python3.5/site-packages/bloom/commands/release.py", line 302, in get_repo_uri
    distribution_file = get_distribution_file(distro)
  File "/usr/lib64/python3.5/site-packages/bloom/commands/release.py", line 251, in get_distribution_file
    files = rosdistro.get_distribution_files(get_index(), distro)
  File "/usr/lib64/python3.5/site-packages/bloom/commands/release.py", line 212, in get_index
    _rosdistro_index = rosdistro.get_index(get_index_url())
  File "/usr/lib64/python3.5/site-packages/bloom/commands/release.py", line 187, in get_index_url
    data = gh.get_branch(owner, repo, branch)
  File "/usr/lib64/python3.5/site-packages/bloom/github.py", line 169, in get_branch
    return json.loads(resp.read())
  File "/usr/lib64/python3.5/json/__init__.py", line 312, in loads
    s.__class__.__name__))
TypeError: the JSON object must be str, not 'bytes'
```

I see there are errors in the CI build, I can try to address them. Looks like they're PEP 8 style issues.

**Comment by nuclearsandwich** [comment]
Thanks for fixing the linter. It's been a busy week but I'll try and test these changes out Monday. 👍

**Comment by nuclearsandwich** [comment]
I still need to test that this under python2.

**Comment by nuclearsandwich** [comment]
@gemarcano was this pull request sufficient to allow you to run bloom-release fully _and_ allow bloom to create a pull request?

I was testing this under python3 and ran into issues that may only surface when doing ROS 2 stuff, which is slightly different

```
Traceback (most recent call last):
  File "/home/steven/osrf/bloom2env/bloom/bloom/commands/release.py", line 1241, in perform_release
    track, repository, distro, interactive, override_release_repository_url
  File "/home/steven/osrf/bloom2env/bloom/bloom/commands/release.py", line 755, in open_pull_request
    updated_distribution_file = generate_ros_distro_diff(track, repository, distro, override_release_repository_url)
  File "/home/steven/osrf/bloom2env/bloom/bloom/commands/release.py", line 585, in generate_ros_distro_diff
    udiff_file = os.path.join(temp_dir, repository + '-' + version + '.patch')
TypeError: Can't convert 'bytes' object to str implicitly

Failed to open pull request: TypeError - Can't convert 'bytes' object to str implicitly
```

File: bloom/github.py [hunk 1]
```diff
@@ -99,6 +99,16 @@ def do_github_post_req(path, data=None, auth=None, site='api.github.com'):
     return response
 
 
+def json_loads(resp):
```
Comment (nuclearsandwich) [inline bloom/github.py]: @gemarcano could you please add a docstring to this function detailing its utility as a wrapper for python2/3 compatibility?

**Comment by nuclearsandwich** [comment]
If @gemarcano has a chance to add the docstring then this will be good to go.

**Comment by nuclearsandwich** [comment]
I added a docstring, as long as it lints, this will be good to merge.

**Pull request merged**